File: algorithms/Maximum Product of Word Lengths/Maximum Product of Word Lengths.py
class Solution:
    def maxProduct(self, words: List[str]) -> int:
        def check(w1, w2):
            for i in w1:
                if i in w2: 
                    return False
            return True
        
        n = len(words)   
        Max = 0
        for i in range(n):
            for j in range(i+1, n):
                if check(words[i], words[j]):
                    Max = max(Max, len(words[i])*len(words[j]))      
        return Max

File: algorithms/M/Maximum Product of Word Lengths/Maximum Product of Word Lengths.cpp
class Solution {
public:
    int maxProduct(vector<string>& words) {
        vector<unordered_set<char>> st;
        int res =0;
        for(string s : words){
            unordered_set<char> temp;
            for(char c : s){
                temp.insert(c);
            }
            st.push_back(temp);
        }
        for(int i = 0;i<words.size();i++){
            for(int j =0;j<i;j++){
                bool flag = false;
                for(auto val : st[i]){
                    if(st[j].find(val) != st[j].end()){
                        flag = true;
                        break;
                    }
                }
                if(flag == false) {
                    int prd = words[i].length() * words[j].length();
                    res =max(res, prd);
                }
            }
        }
        return res;
    }
};
